### PR TITLE
[4.0] Changing UI for com_mails list view

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #26489.

### Summary of Changes
The UI of the mail template manager was not intuitive and a concern for a beta release. This attempts to change this to something that people can understand better. The changes are:
- The language column only displays the languages for which a customised template exists. Should find a better title for that column, I guess...
- To create or edit a template, you click on the title and get a dropdown to select the language that you want to edit.

### Testing Instructions
This is a proposal, so I'm looking for feedback on the useability of this.

### CODE REVIEW

**Comment by Hackwar** [comment]
@brianteeman, @infograf768 Your opinions about this? As I wrote in other areas, this should indeed be improved and it would be a pity if we have to drop this whole feature for the 4.x series.

**Comment by brianteeman** [comment]
@Hackwar its much better but there are still some issues to address

1. I think it would be useful to know if there are languages where the template doesn't exist yet - so I guess another column?

2. As there is no template for language ALL that option needs to be removed from the filter

3. On the edit mail template there should be a visual indicator what language you are creating a template for - perhaps
image

3. It is not clear to me from the UI when you need to create the template for a language AND if you only create the template for one language will that be used for all languages or will the other languages use the core

**Comment by brianteeman** [comment]
OR would it be helpful to have the ability to create a template for Language - ALL

**Comment by Hackwar** [comment]
>     1. I think it would be useful to know if there are languages where the template doesn't exist yet - so I guess another column?

The idea is, that the dropdown says "Create template for language X" if there is none or "Edit template for language X" if there is one. 

>     2. As there is no template for language ALL that option needs to be removed from the filter

Yes.

>     3. On the edit mail template there should be a visual indicator what language you are creating a template for - perhaps

Right now I want to concentrate on the list view. The edit view is another issue and I'd like to handle that as a different PR as well. But I will keep this in mind. You are right.

>     4. It is not clear to me from the UI when you need to create the template for a language AND if you only create the template for one language will that be used for all languages or will the other languages use the core

It will always use the default translation strings given or whatever the devs insert in the field of the default entry for that template ID, unless you created a customisation for a specific language. It is not going to be used for other languages.

Generally, what would an ALL template look like? I can't really come up with something that wouldn't be covered by the translation strings already...

**Comment by brianteeman** [comment]
1. The dropdown is fine and it works well but it won't work well when you have a large number of templates to configure hence the additional suggestion of the column
2.
3. OK - its an easy fix

4. 
> It will always use the default translation strings given or whatever the devs insert in the field of the default entry for that template ID, unless you created a customisation for a specific language. It is not going to be used for other languages.

Let me ask that a different way. On a site with English, French and German and I _only_ create a French template - what will an english or german user see?

**Comment by Hackwar** [comment]
>     1. The dropdown is fine and it works well but it won't work well when you have a large number of templates to configure hence the additional suggestion of the column

I'm not sure I follow you here... Basically, the dropdown would contain a row for every language, so unless you have 10+ languages in your site, you are pretty fine...

> > It will always use the default translation strings given or whatever the devs insert in the field of the default entry for that template ID, unless you created a customisation for a specific language. It is not going to be used for other languages.
> 
> Let me ask that a different way. On a site with English, French and German and I _only_ create a French template - what will an english or german user see?

If you don't have a customised template for a language, the system falls back onto what is stored in the default template for that template id. That normally would be a translation key, which would then again be in some language file, which hopefully is already loaded.

**Comment by brianteeman** [comment]
> I'm not sure I follow you here... Basically, the dropdown would contain a row for every language, so unless you have 10+ languages in your site, you are pretty fine...

That is fine when you are only dealing with one template. Now imagine you have twenty templates and five languages. By adding the extra column to indicate the languages on the site that do _not_ have a customisation then you can quickly see without having to click on each of the 20 dropdowns

**Comment by Hackwar** [comment]
Hmm, I would argue that it is easier to see when we don't have that second column. Let me break into a large and very pseudo-scientific explanation, which I'm then going to cut short with another argument. 😛 

The human eye can see rythm changes rather well. I would assume that it is rather easier to see if in a group of 5 icons two are missing, than when you have 3 icons in column one and 2 icons in column two. (With a site that has 5 languages installed)
Now for the argument to cut all this short: We don't have this second column for language article associations either, so why do we need it here?

**Comment by brianteeman** [comment]
I will counter that with an image of the multilingual associations component ui
#
image

**Comment by Hackwar** [comment]
damn, you got me. Fair enough.

**Comment by brianteeman** [comment]
:)

**Comment by brianteeman** [comment]
@hackwar I made a pr for this part #27723 
image

**Comment by Hackwar** [comment]
Second columm has been added. I'm open for a better wording...

**Comment by brianteeman** [comment]
thanks - i cant think of anything else right now for the text

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on eadfa2d2da5ecafc08fe627704398330e6e0c744<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27704">issues.joomla.org/tracker/joomla-cms/27704</a>.</sub>

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on eadfa2d2da5ecafc08fe627704398330e6e0c744<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27704">issues.joomla.org/tracker/joomla-cms/27704</a>.</sub>

File: administrator/components/com_mails/tmpl/templates/default.php [hunk 1]
```diff
@@ -40,8 +40,11 @@
 								<th scope="col" style="width:15%" class="d-none d-md-table-cell">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_MAILS_HEADING_COMPONENT', 'a.component', $listDirn, $listOrder); ?>
 								</th>
-								<th scope="col" style="width:15%"  class="d-md-table-cell">
-									<?php echo Text::_('COM_MAILS_HEADING_LANGUAGES'); ?>
+								<th scope="col" style="width:10%"  class="d-md-table-cell">
```
Comment (Quy) [inline administrator/components/com_mails/tmpl/templates/default.php]:
```suggestion
								<th scope="col" style="width:10%" class="d-md-table-cell">
```

File: administrator/components/com_mails/tmpl/templates/default.php [hunk 2]
```diff
@@ -40,8 +40,11 @@
...
+									<?php echo Text::_('COM_MAILS_HEADING_TEMPLATES_FOR_LANGUAGES'); ?>
+								</th>
+								<th scope="col" style="width:10%"  class="d-md-table-cell">
```
Comment (Quy) [inline administrator/components/com_mails/tmpl/templates/default.php]:
```suggestion
								<th scope="col" style="width:10%" class="d-md-table-cell">
```

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27704">issues.joomla.org/tracker/joomla-cms/27704</a>.</sub>

**Pull request merged**